vector-search-improve-performance.md

```diff
@@ -0,0 +1,52 @@
+
+</CustomContent>
+
+> **Note:**
+>
+> The vector search feature is only available for TiDB Self-Managed clusters and [TiDB Cloud Serverless](/tidb-cloud/select-cluster-tier.md#tidb-cloud-serverless) clusters.
+
+## Add vector search index for vector columns
+
+The [vector search index](https://docs.pingcap.com/tidbcloud/vector-search-index) dramatically improves the performance of vector search queries, usually by 10x or more, with a trade-off of only a small decrease of recall rate.
+
+## Ensure vector indexes are fully built
+
+> **Note**
+>
+> This practice is only applicable to [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#tidb-serverless) clusters.
+
+Vector indexes are built asynchronously. Until all vector data is indexed, vector search performance is suboptimal. To check the index build progress, see [View index build progress](https://docs.pingcap.com/tidbcloud/vector-search-index#view-index-build-progress).
+
+## Reduce vector dimensions or shorten embeddings
+
+The computational complexity of vector search indexing and queries increases significantly as the dimension of vectors grows, necessitating more floating point comparisons.
+
+To optimize performance, consider reducing the vector dimensions whenever feasible. This usually needs switching to another embedding model. When switching models, you need to evaluate the impact of the model change on the accuracy of vector queries.
+
+Certain embedding models like OpenAI `text-embedding-3-large` support [shortening embeddings](https://openai.com/index/new-embedding-models-and-api-updates/), which removes some numbers from the end of vector sequences without losing the embedding's concept-representing properties. You can also use such an embedding model to reduce the vector dimensions.
+
+## Exclude vector columns from the results
+
+Vector embedding data are usually large and only used during the search process. By excluding vector columns from the query results, you can greatly reduce the amount of data transferred between the TiDB server and your SQL client, thereby improving query performance.
+
+To exclude vector columns, explicitly list the columns you want to retrieve in the `SELECT` clause, instead of using `SELECT *` to retrieve all columns.
+
+## Warm up the index
+
+When accessing an index that has never been used or has not been accessed for a long time (cold access), TiDB needs to load the entire index from S3 or disk (instead of from memory). This process takes some time and often results in higher query latency. Additionally, if there are no SQL queries for an extended period (for example, several hours), the computing resources is reclaimed, causing subsequent access to become cold access.
```

Comment:
```suggestion
When accessing an index that has never been used or has not been accessed for a long time (cold access), TiDB needs to load the entire index from cloud storage or disk (instead of from memory). This process takes time and often results in higher query latency. Additionally, if there are no SQL queries for an extended period (for example, several hours), computing resources are reclaimed, causing subsequent access to become cold access.
```